src/network/networkd-ndisc.c

```diff
@@ -880,6 +1002,14 @@ static int ndisc_router_handler(Link *link, sd_ndisc_router *rt) {
                 return 0;
         }
 
+        r = sd_ndisc_router_get_timestamp(rt, CLOCK_BOOTTIME, &timestamp_usec);
+        if (r < 0)
+                return r;
+
+        r = ndisc_drop_outdated(link, timestamp_usec);
```

Comment:
Since this version of the PR calls `ndisc_drop_outdated()` as soon as any lifetime expires, we don't still need to call it here, do we? I think we can also get rid of `ndisc_vacuum()` entirely for the same reason. Let me know if I'm missing something,  though.

Reply:
Thank you for the comments!

> Since this version of the PR calls `ndisc_drop_outdated()` as soon as any lifetime expires, we don't still need to call it here, do we?

May be, but may not be. The timer event source has `accuracy` parameter, which defaults to 250ms. So, there still exists possibility that we receive a RA when some information is outdated. Hopefully, this is not a heavy task, at least when there is no host that sends spam RA.

> I think we can also get rid of `ndisc_vacuum()` entirely for the same reason. Let me know if I'm missing something, though.

Yeah. Good point. I will submit further follow-ups.